consensus/src/consensus_observer/observer.rs

```diff
@@ -475,21 +525,52 @@ impl ConsensusObserver {
             block_round,
         );
 
-        // TODO: verify the block payload!
+        // Verify the block payload digests
+        if let Err(error) = block_payload.verify_payload_digests() {
+            error!(
+                LogSchema::new(LogEntry::ConsensusObserver).message(&format!(
+                    "Failed to verify block payload digests! Ignoring block: {:?}. Error: {:?}",
+                    block_payload.block, error
+                ))
+            );
+            return;
+        }
+
+        // If the payload is for the current epoch, verify the proof signatures
+        let epoch_state = self.get_epoch_state();
+        let verified_payload = if block_epoch == epoch_state.epoch {
```

Comment:
not sure if it's too much complexity to handle messages across epoch. we can state sync after joining the new epoch and then buffer messages from the same epoch

Reply:
Yeah, I originally started with this, but it turns out to be pretty bad when running at max load across the epoch boundaries. The nodes fall behind and take a while to catch up (spiking the e2e latency). But, we can see how this performs and maybe simplify it if we feel the need (epoch changes are more frequent in our tests) 😄